### PR TITLE
feat(iam): ANA-1450 Add create account service

### PR DESCRIPTION
**Pull request by avattipalli**
Jira: [ANA-1450](https://jira.bigcommerce.com/browse/ANA-1450)

## What? Why?
This change is for the tests to be able to create an account in the absence of an existing account they could use to run any e2e tests. The service itself calls the internal API mentioned below to create an account. In addition to that I have also added API call within the codeceptjs which gives us an idea on how to make HTTP requests if we do not have an internal API already doing that for us.

For testing this I added [nock](https://github.com/nock/nock) for intercepting HTTP requests and mock the responses to test the account service creation.

Thanks to @aaron-humerickhouse for pointing out https://github.com/bigcommerce-labs/bc-store-setup-node which can be used to create account. 

## How was it tested?
Tested locally

- [ ] Cloud Dev end-to-end - Describe Steps
- [x] Additional test cases (Optional)
- [ ] Integration - Describe Steps (Optional)

POL:
<br/>
Test Results
<br/>

## How can this be undone?
1. Revert this PR

ping @bigcommerce/billing-iam-testing

### CODE REVIEW

File: codecept.conf.js [hunk 1]
```diff
@@ -3,18 +3,30 @@ require('import-export');
 
 setHeadlessWhen(process.env.HEADLESS);
 
+const STAGING_BMP_URL = 'https://manage.staging.zone/';
+const INTEGRATION_BMP_URL = 'https://manage.integration.zone/';
```
Comment (aaron-humerickhouse) [inline codecept.conf.js:7]: should this be defined in a .env file?

File: src/config/data/account-data.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+const faker = require('faker');
+
+module.exports = Object.freeze({
+    AccountValues: {
```
Comment (aaron-humerickhouse) [inline src/config/data/account-data.js:4]: does it need this top level AccountValues?  If it's removed, it cleans up services/create-account

**Comment by aaron-humerickhouse** [comment]
Add unit tests for the service
Show proof of it working

**Comment by avattipalli** [comment]
While adding tests I noticed that codeceptjs doesn't provide a great mocking mechanism to make things easy. Although we can mock requests using its httpmock services, the functionality it provides is limited. For example in this call, 

`this.mockRequest('POST', ACCOUNTS_ENDPOINT, create_account_payload, this._buildHeaders());`

it mocks a post call but it doesn't return anything for us to verify an expected response. For this reason, I had to add other mock packages and assertion libraries to be able to test the service.

**Comment by aaron-humerickhouse** [comment]
So end to end tests shouldn't have mocks, ideally.  There are edge cases where we may want to mock out third party services.

Unit testing our services I would dig into https://github.com/axios/moxios or mocking using jest https://jestjs.io/docs/en/tutorial-async.  Do not unit test using codeceptjs.  That's not the right tool.

Functional or End to End where we really do want to mock out services, look at Pact: https://docs.pact.io/ and only use the consumer half or something like MSL: http://finraos.github.io/MSL/index.html

**Comment by avattipalli** [comment]
> So end to end tests shouldn't have mocks, ideally. There are edge cases where we may want to mock out third party services.
> 
> Unit testing our services I would dig into https://github.com/axios/moxios or mocking using jest https://jestjs.io/docs/en/tutorial-async. Do not unit test using codeceptjs. That's not the right tool.
> 
> Functional or End to End where we really do want to mock out services, look at Pact: https://docs.pact.io/ and only use the consumer half or something like MSL: http://finraos.github.io/MSL/index.html

Rightly said. We need to stay away from mocks as much as we can here and also not rely on codeptjs much. I have written tests where I mock the service using the end point url but backed out as it wasn't actually testing the code written. Probably that is somewhat expected but may be I was overthinking a bit on how to test this ideally. I'd like to bring this up in our meeting offline and discuss more.

File: codecept.conf.js [hunk 3]
```diff
@@ -1,35 +1,60 @@
 const { setHeadlessWhen } = require('@codeceptjs/configure');
 require('import-export');
+require('dotenv').config();
 
 setHeadlessWhen(process.env.HEADLESS);
 
+const HOST = process.env.ENVIRONMENT === 'staging' ? process.env.STAGING_BMP_URL : process.env.INTEGRATION_BMP_URL;
```
Comment (j05h) [inline codecept.conf.js:7]: Are we building this tool to only work against these two environments?

Comment (avattipalli) [reply codecept.conf.js:7]: Yes, we wanted to run the tests in integration and staging environment but not in production. I can tweak that to point to `login.bigcommerce.bcdev` to be able to run the tests locally if it isn't in either of those two env.

Comment (aaron-humerickhouse) [reply codecept.conf.js:7]: This should likely be determined at the repository level and not at the library level.  It may be worth preventing execution on production.

A discussion worth having.

File: package.json [hunk 4]
```diff
@@ -16,26 +16,29 @@
...
     "codeceptjs": "^2.6.5",
-    "codeceptjs-httpmock": "0.0.7",
     "axios": "latest",
     "@wdio/cli": "^6.1.12",
     "@wdio/local-runner": "^5.22.1",
     "@wdio/selenium-standalone-service": "^5.16.10",
-    "webdriverio": "^6.1.0"
+    "webdriverio": "^6.1.0",
+    "bc-store-setup-node": "git+ssh://git@github.com/bigcommerce-labs/bc-store-setup-node.git#master",
+    "@jest/globals": "latest"
```
Comment (aaron-humerickhouse) [inline package.json:25]: jest would be a dev dependency

File: test/create-account-test.js [hunk 5]
```diff
@@ -1,8 +1,20 @@
 const {accountCreation} = inject();
 const expect = require('chai').expect;
+import axios from 'axios';
+import moxios from 'moxios';
 
 Feature('create account service tests');
 
+const axiosInstance = axios.create();
```
Comment (aaron-humerickhouse) [inline test/create-account-test.js:8]: If moxios requires an axios instance, this is the wrong tool.  Maybe something like Nock: https://github.com/nock/nock.

We need to be testing the end result of 3rd party code.  That means we don't technically have an axios to pass into moxios

File: test/create-account-test.js [hunk 6]
```diff
@@ -19,3 +31,14 @@ Scenario('Test create account fails with network error', async() => {
     expect(res).to.be.an.instanceof(Error, 'Network Error');
 });
 
+Scenario('Test create account using moxios and stubbing out request', async() => {
```
Comment (aaron-humerickhouse) [inline test/create-account-test.js:34]: this does nothing because your making the http request in the test.  You need to exercise bc-store-setup-node's library and test that.

File: src/config/data/account-data.js [hunk 7]
```diff
@@ -0,0 +1,18 @@
+const faker = require('faker');
+
+module.exports = Object.freeze({
+    email: `auto-${new Date().getTime()}@bigcommerce.com`,
```
Comment (MarkJCram) [inline src/config/data/account-data.js:4]: Is there any way to pass the test name/number as part of the email address?  Would make it significantly easier to dig into test failures, especially if multiple parallel tests were running that used emails

Comment (avattipalli) [reply src/config/data/account-data.js:4]: I kinda see that we can have a better name around the email for test account. I'll look into see if there's a way to extract runner name or something close.

Comment (aaron-humerickhouse) [reply src/config/data/account-data.js:4]: We will need to make sure that we have unique emails.  Time doesn't always get us that and test name certainly won't.  The combination should though.

**Comment by avattipalli** [comment]
> I would expect that unit tests are run with something like Jest and not CodeceptJS (acceptance test framework). Splitting the unit tests off will also remove the chance that the tests get run with any functional tests.

As discussed offline, created a separate task to work on establishing jest for unit testing. (QE-162)[https://jira.bigcommerce.com/browse/QE-162]

File: src/config/data/account-data.js (hunk 7 above)
Comment (avattipalli) [reply src/config/data/account-data.js]: Updated the email format to contain <TestName>-<Hours+Minutes>@bigcommerce.com. Updated screenshot of the test run has the format in it
